pkg/controller/factory/base_controller.go

```diff
@@ -202,7 +204,12 @@ func (c *baseController) runWorker(queueCtx context.Context) {
 // reconcile wraps the sync() call and if operator client is set, it handle the degraded condition if sync() returns an error.
 func (c *baseController) reconcile(ctx context.Context, syncCtx SyncContext) error {
 	err := c.sync(ctx, syncCtx)
-	return c.reportDegraded(ctx, err)
+	degradedErr := c.reportDegraded(ctx, err)
```

Comment:
can you split this out to separate PR? I can approve it right now

Reply:
Not sure what to split. Handling of `management.IsOperatorRemovable()`? IMO it belongs here, where the other controllers are updated too.